samples/batch/transcription-enabled-storage/CreateByTimerFunction/CreateTranscriptionByTimer.cs

```diff
@@ -0,0 +1,348 @@
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+// </copyright>
+
+namespace CreateByTimerFunction
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Connector;
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.ServiceBus.Core;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public static class CreateTranscriptionByTimer
+    {
+        private const int MaxMessagesPerRequest = 500;
+
+        private const int MaxFilesPerTranscriptionJob = 500;
+
+        private static DateTime StartDateTime;
+
+        private static ILogger Logger;
+
+        [FunctionName("CreateTranscriptionByTimer")]
+        public static async Task Run([TimerTrigger("0 */2 * * * *")] TimerInfo myTimer, ILogger log)
+        {
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            if (myTimer == null)
+            {
+                throw new ArgumentNullException(nameof(myTimer));
+            }
+
+            Logger = log;
+            StartDateTime = DateTime.UtcNow;
+            log.LogInformation($"C# Timer trigger function v3 executed at: {StartDateTime}. Next occurrence on {myTimer.Schedule.GetNextOccurrence(StartDateTime)}.");
+
+            if (!string.IsNullOrEmpty(CreateTranscriptionEnvironmentVariables.SecondaryLocale) && !CreateTranscriptionEnvironmentVariables.SecondaryLocale.Equals("None", StringComparison.OrdinalIgnoreCase))
+            {
+                log.LogWarning($"Language identification is not supported for timer-based transcription creation.");
+            }
+
+            var receiver = new MessageReceiver(CreateTranscriptionEnvironmentVariables.AzureServiceBus, "create_transcription_queue");
+
+            var totalServiceBusMessages = new List<Message>();
+
+            var serviceBusMessagesRemaining = true;
+            log.LogInformation("Pulling messages from queue...");
+            do
+            {
+                var messages = await receiver.ReceiveAsync(MaxMessagesPerRequest, TimeSpan.FromSeconds(20)).ConfigureAwait(false);
+
+                if (messages == null)
+                {
+                    log.LogInformation("No remaining messages found in queue.");
+                    break;
+                }
+
+                foreach (var message in messages)
+                {
+                    await receiver.CompleteAsync(message.SystemProperties.LockToken).ConfigureAwait(false);
+                    totalServiceBusMessages.Add(message);
+                }
+
+                if (messages.Count < MaxMessagesPerRequest)
+                {
+                    serviceBusMessagesRemaining = false;
```

Comment:
I do not understand the logic here: if messages.Count >= MaxMessagesPerRequest, you want to continue the loop?

Reply:
as discussed